### PR TITLE
PADV-944 Add redux in students and courses pages

### PR DESCRIPTION
**Pull request by AuraAlba**
### Ticket
https://agile-jira.pearson.com/browse/PADV-944

### Description
Manage unique state with redux to shared state between components

### Changes made
Define redux actions and reducer for couses and students pages
Import redux store in other pages
Add and update tests

### How to test
Clone the Institution Portal MFE
Run npm install.
Run npm run start
Go to http://localhost:1980/

### CODE REVIEW

File: src/features/Students/data/slice.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+const initialState = {
+  table: {
+    currentPage: 1,
+    data: [],
+    status: RequestStatus.LOADING,
+    error: null,
+    numPages: 0,
+    count: 0,
+  },
+  courses: {
```
Comment (anfbermudezme) [inline src/features/Students/data/slice.js:14]: courses, classes and filters have the same initialState, is there some way to unify it? @AuraAlba

File: src/features/Students/data/slice.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    data: [],
+    status: RequestStatus.LOADING,
+    error: null,
+  },
+  classes: {
+    data: [],
+    status: RequestStatus.LOADING,
+    error: null,
+  },
+  filters: {},
```
Comment (anfbermudezme) [inline src/features/Students/data/slice.js:24]: Could you please move this to the end of the inititalState. @AuraAlba

File: src/features/Students/StudentsTable/_test_/index.test.jsx [hunk 3]
```diff
@@ -44,7 +56,9 @@ describe('Student Table', () => {
     ];
 
     const component = render(
-      <StudentsTable data={data} count={data.length} columns={getColumns()} />,
+      <Provider store={store}>
+        <StudentsTable data={data} count={data.length} columns={getColumns()} />
+      </Provider>,
     );
 
     // Check if the table rows are present
```
Comment (anfbermudezme) [inline src/features/Students/StudentsTable/_test_/index.test.jsx:64]: Is this inline comment necessary? @AuraAlba

File: src/features/Instructors/InstructorsFilters/index.jsx [hunk 4]
```diff
@@ -43,7 +43,9 @@ const InstructorsFilters = ({ resetPagination }) => {
     const formJson = Object.fromEntries(formData.entries());
     dispatch(updateFilters(formJson));
     try {
-      dispatch(fetchInstructorsData(currentPage, formJson));
+      const initialPage = 1;
```
Comment (anfbermudezme) [inline src/features/Instructors/InstructorsFilters/index.jsx:46]: What do you think about having a module dedicated to constants? @AuraAlba

File: src/features/Courses/CoursesPage/index.jsx [hunk 5]
```diff
@@ -1,64 +1,34 @@
...
-};
+import { updateCurrentPage } from 'features/Courses/data/slice';
+import { fetchCoursesData } from 'features/Courses/data/thunks';
 
 const CoursesPage = () => {
   const stateInstitution = useSelector((state) => state.main.institution.data);
-  const [state, dispatch] = useReducer(reducer, initialState);
+  const stateCourses = useSelector((state) => state.courses);
+  const dispatch = useDispatch();
   const [currentPage, setCurrentPage] = useState(1);
```
Comment (anfbermudezme) [inline src/features/Courses/CoursesPage/index.jsx:18]:
```suggestion
  const [currentPage, setCurrentPage] = useState(initialPage);
```

File: src/features/Courses/CoursesFilters/index.jsx [hunk 6]
```diff
@@ -1,46 +1,59 @@
...
 import { Select, Button } from 'react-paragon-topaz';
 import { logError } from '@edx/frontend-platform/logging';
 import PropTypes from 'prop-types';
 
+import { updateFilters, updateCurrentPage } from 'features/Courses/data/slice';
+import { fetchCoursesData } from 'features/Courses/data/thunks';
+
 const CoursesFilters = ({
-  fetchData, resetPagination, dataCourses, setFilters,
+  resetPagination,
```
Comment (anfbermudezme) [inline src/features/Courses/CoursesFilters/index.jsx:13]:
```suggestion
  const CoursesFilters = ({ resetPagination }) => {
```

File: src/features/Courses/CoursesFilters/_test_/index.test.jsx [hunk 7]
```diff
@@ -71,19 +97,16 @@ describe('InstructorsFilters Component', () => {
...
     const resetPagination = jest.fn();
     const { getByText, getByTestId } = render(
-      <CoursesFilters
-        fetchData={fetchData}
-        resetPagination={resetPagination}
-        setFilters={mockSetFilters}
-        dataCourses={dataCourses}
-      />,
+      <Provider store={store}>
+        <CoursesFilters
```
Comment (anfbermudezme) [inline src/features/Courses/CoursesFilters/_test_/index.test.jsx:106]:
```suggestion
        <CoursesFilters resetPagination={resetPagination} />
```

File: src/features/Courses/CoursesFilters/_test_/index.test.jsx [hunk 8]
```diff
@@ -23,42 +33,58 @@ jest.mock('react-select', () => function reactSelect({ options, valueR, onChange
...
```
Comment (anfbermudezme) [inline src/features/Courses/CoursesFilters/_test_/index.test.jsx:80]:
```suggestion
        <CoursesFilters resetPagination={resetPagination} />
```

**Review by anfbermudezme — CHANGES_REQUESTED** [review]

File: src/features/Students/data/slice.js (hunk 1 above)
Comment (AuraAlba) [reply src/features/Students/data/slice.js:14]: I created a constat and import it in the states with the same structure

File: src/features/Students/data/slice.js (hunk 2 above)
Comment (AuraAlba) [reply src/features/Students/data/slice.js]: done

File: src/features/Students/StudentsTable/_test_/index.test.jsx (hunk 3 above)
Comment (AuraAlba) [reply src/features/Students/StudentsTable/_test_/index.test.jsx]: Not. I removed them

File: src/features/Instructors/InstructorsFilters/index.jsx (hunk 4 above)
Comment (AuraAlba) [reply src/features/Instructors/InstructorsFilters/index.jsx]: We already have a constants file, so I moved it there

File: src/features/Courses/CoursesPage/index.jsx (hunk 5 above)
Comment (AuraAlba) [reply src/features/Courses/CoursesPage/index.jsx]: done

File: src/features/Courses/CoursesFilters/index.jsx (hunk 6 above)
Comment (AuraAlba) [reply src/features/Courses/CoursesFilters/index.jsx]: done

File: src/features/Courses/CoursesFilters/_test_/index.test.jsx (hunk 8 above)
Comment (AuraAlba) [reply src/features/Courses/CoursesFilters/_test_/index.test.jsx]: done

**Review by anfbermudezme — APPROVED** [review]

**Pull request merged**